packages/plugin/src/ChartPlugin.js

```diff
@@ -1,103 +1,21 @@
 import React, { useRef, useCallback, useEffect } from 'react'
 import PropTypes from 'prop-types'
-import i18n from '@dhis2/d2-i18n'
-import {
-    isYearOverYear,
-    isSingleValue,
-    createVisualization,
-} from '@dhis2/analytics'
-
-import {
-    apiFetchAnalytics,
-    apiFetchAnalyticsForYearOverYear,
-} from './api/analytics'
-import { getOptionsForRequest } from './modules/options'
-import { computeGenericPeriodNames } from './modules/analytics'
-
-const getRequestOptions = (visualization, filters) => {
-    const options = getOptionsForRequest().reduce((map, [option, props]) => {
-        // only add parameter if value !== default
-        if (
-            visualization[option] !== undefined &&
-            visualization[option] !== props.defaultValue
-        ) {
-            map[option] = visualization[option]
-        }
-
-        return map
-    }, {})
-
-    // interpretation filter
-    if (filters.relativePeriodDate) {
-        options.relativePeriodDate = filters.relativePeriodDate
-    }
-
-    // global filters
-    // userOrgUnit
-    if (filters.userOrgUnit && filters.userOrgUnit.length) {
-        const ouIds = filters.userOrgUnit.map(
-            ouPath => ouPath.split('/').slice(-1)[0]
-        )
-
-        options.userOrgUnit = ouIds.join(';')
-    }
-
-    return options
-}
-
-const fetchData = async ({ visualization, filters, d2, forDashboard }) => {
-    const options = getRequestOptions(visualization, filters)
-
-    const extraOptions = {
-        dashboard: forDashboard,
-        noData: { text: i18n.t('No data') },
-    }
-
-    if (isYearOverYear(visualization.type)) {
-        const {
-            responses,
-            yearlySeriesLabels,
-        } = await apiFetchAnalyticsForYearOverYear(d2, visualization, options)
-
-        return {
-            responses,
-            extraOptions: {
-                ...extraOptions,
-                yearlySeries: yearlySeriesLabels,
-                xAxisLabels: computeGenericPeriodNames(responses),
-            },
-        }
-    }
-
-    const responses = await apiFetchAnalytics(d2, visualization, options)
-
-    return {
-        responses,
-        extraOptions,
-    }
-}
+import { isSingleValue, createVisualization } from '@dhis2/analytics'
 
 const ChartPlugin = ({
     visualization,
-    filters,
-    id,
+    responses,
+    extraOptions,
+    // legendSets,
```

Comment:
We should enable this already as @martinkrulltott is implementing legend set for Gauge.

Reply:
@martinkrulltott you can uncomment this line and the one in propTypes when we use legends in ChartPlugin, it is passed by the caller.  It breaks the `no-unused-variables` ESLint rule to leave it here now